core/services/keeper/orm.go

```diff
@@ -0,0 +1,109 @@
+package keeper
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+const gasBuffer = int32(200_000)
+
+func NewORM(db *gorm.DB) KeeperORM {
+	return KeeperORM{
+		DB: db,
+	}
+}
+
+type KeeperORM struct {
+	DB *gorm.DB
+}
+
+func (korm KeeperORM) Registries() (registries []Registry, _ error) {
```

Comment:
All of these orm methods should take a context and use korm.DB.WithContext(ctx) / sqlDB.ExecContext since they are not part of larger transactions using postgres.GormTransaction

Reply:
and the context should have some sane deadline, perhaps postgres.DefaultQueryCtx